File: web/src/api/kubernetes/config/secret.js
import service from '@/utils/request'

// 获取secret list
export const getSecretList = (data) => {
  return service({
    url: '/secret/getSecretList',
    method: 'post',
    data
  })
}

// 获取secret编排
export const getSecretRaw = (data) => {
  return service({
    url: '/secret/getSecretRaw',
    method: 'post',
    data
  })
}

// 获取secret详情
export const getSecretDetail = (data) => {
  return service({
    url: '/secret/getSecretDetail',
    method: 'post',
    data
  })
}

// 删除secret
export const deleteSecret = (data) => {
  return service({
    url: '/secret/deleteSecret',
    method: 'post',
    data
  })
}

File: server/router/kubernetes/secret.go
package kubernetes

import (
	"github.com/gin-gonic/gin"

	v1 "github.com/pddzl/kubeops/server/api/v1"
	"github.com/pddzl/kubeops/server/middleware"
)

type SecretRouter struct{}

func (s *SecretRouter) InitSecretRouter(Router *gin.RouterGroup) {
	secretRouter := Router.Group("secret").Use(middleware.OperationRecord())
	secretRouterWithoutRecord := Router.Group("secret")
	secretApi := v1.ApiGroupApp.KubernetesApiGroup.SecretApi
	{
		secretRouter.POST("getSecretRaw", secretApi.GetSecretRaw)       // 获取secret编排
		secretRouter.POST("getSecretDetail", secretApi.GetSecretDetail) // 获取secret详情
	}
	{
		secretRouterWithoutRecord.POST("getSecretList", secretApi.GetSecretList) // 获取所有secret
	}
}
